src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -501,10 +524,15 @@ public bool ContainsSyntaxTree(SyntaxTree syntaxTree)
 
         protected abstract bool CommonContainsSyntaxTree(SyntaxTree? syntaxTree);
 
+        /// <summary>
+        /// Optional semantic model provider for this compilation.
+        /// </summary>
+        internal SemanticModelProvider? SemanticModelProvider { get; }
+
         /// <summary>
         /// The event queue that this compilation was created with.
         /// </summary>
-        internal readonly AsyncQueue<CompilationEvent>? EventQueue;
+        internal AsyncQueue<CompilationEvent>? EventQueue { get; }
```

Comment:
Why did this need to be changed in this PR?

Reply:
So I first ended up copy-pasting this field to add the `SemanticModelProvider` and @jaredpar correctly pointed out it could be a readonly property instead - I decided to change both the existing field and newly added ones to properties.